src/common/SurgeStorage.cpp

```diff
@@ -300,13 +302,43 @@ void SurgeStorage::refreshPatchlistAddDir(bool userDir, string subdir)
       return;
    }
 
-   for (auto& p : fs::directory_iterator(patchpath))
+   /*
+   ** std::filesystem has a recursive_directory_iterator, but between the 
+   ** hand rolled ipmmlementation on mac, expermiental on windows, and 
+   ** ostensibly standard on linux it isn't consistent enough to warrant
+   ** using yet, so build my own recursive directory traversal with a simple
+   ** stack
+   */
+   std::vector<fs::path> alldirs;
+   std::deque<fs::path> workStack;
+   workStack.push_back( patchpath );
+   while (!workStack.empty())
+   {
+       auto top = workStack.front();
+       workStack.pop_front();
+       for (auto &d : fs::directory_iterator( top ))
+       {
+           if (fs::is_directory(d))
+           {
+               alldirs.push_back(d);
+               workStack.push_back(d);
+           }
+       }
+   }
+
+   // We want to knock out /user/foo or c:\\users\\bar\\ so get the length once
```

Comment:
And here are using that style (and different one above).

Reply:
If this was by any means consistent, this should be:

``
/*
** We want to knock out /user/foo or c:\\users\\bar\\ so get the length once.
*/
```
To follow the style inline comment above.

Reply:
See discussion below. But anyway I expanded this to be a multiline comment so it's now in /* style.